allennlp/data/token_indexers/pretrained_transformer_pretokenized_indexer.py

```diff
@@ -0,0 +1,95 @@
+from typing import Dict, List, Tuple
+import logging
+
+from overrides import overrides
+import torch
+
+from allennlp.common.util import pad_sequence_to_length
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.data.tokenizers.token import Token
+from allennlp.data.token_indexers import PretrainedTransformerIndexer, TokenIndexer
+from allennlp.data.token_indexers.token_indexer import IndexedTokenList
+from allennlp.data.tokenizers import PretrainedTransformerTokenizer
+
+logger = logging.getLogger(__name__)
+
+
+@TokenIndexer.register("pretrained_transformer_pretokenized")
+class PretrainedTransformerPretokenizedIndexer(PretrainedTransformerIndexer):
+    """
+    Use this indexer when input comes from pre-tokenized text and therefore
+    `PretrainedTransformerTokenizer` was not used in the dataset loader. It tokenizes each token
+    into wordpieces independently and concatenate them back together. Use it along with
+    `PretrainedTransformerPretokenizedEmbedder`.
+    """
+
+    def __init__(
+        self, model_name: str, namespace: str = "tags", token_min_padding_length: int = 0
+    ) -> None:
+        super().__init__(model_name, namespace, token_min_padding_length)
+        # add_special_tokens=False sicne we don't want wordpieces to be surrounded by special tokens
+        self._allennlp_tokenizer = PretrainedTransformerTokenizer(
```

Comment:
This is because the superclass has a `self._tokenizer` for the huggingface tokenizer, which is also used in this class to `build_inputs_with_special_tokens`, etc.

Reply:
Ah, ok, so, getting rid of the inheritance will clarify that issue quite a bit.